File: backend/src/modules/modalidade/entity/Modalidade.js
class Modalidade {
  constructor({ id_modalidade, nome, descricao, exige_agendamento }) {
    this.id_modalidade = id_modalidade;
    this.nome = nome;
    this.descricao = descricao;
    this.exige_agendamento = exige_agendamento;
  }
  validate() {
    const errors = [];
    if (!this.nome) errors.push('Nome é obrigatório');
    return errors;
  }
}
module.exports = Modalidade;

File: backend/src/main/java/com/vitalativa/entity/Modalidade.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "modalidade")
public class Modalidade {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_modalidade")
    private Integer idModalidade;
    
    @NotBlank(message = "Nome é obrigatório")
    private String nome;
    
    private String descricao;
    
    @Column(name = "exige_agendamento")
    private Boolean exigeAgendamento;
    
    public Modalidade() {}
    
    public Integer getIdModalidade() { return idModalidade; }
    public void setIdModalidade(Integer idModalidade) { this.idModalidade = idModalidade; }
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }
    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }
    public Boolean getExigeAgendamento() { return exigeAgendamento; }
    public void setExigeAgendamento(Boolean exigeAgendamento) { this.exigeAgendamento = exigeAgendamento; }
}